**Ejercicio 1**

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?   sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipóParatesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

Hipotesis Nula: No hay diferencia en la pérdida de peso promedio entre las dos dietas.
Hipótesis alternativa: Sí hay una diferencia en la pérdida de peso promedio

In [ ]:
import scipy.stats as stats

diet_1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
diet_2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]
t_value= stats.ttest_ind(diet_1, diet_2, equal_var=False)

print(t_value)

TtestResult(statistic=np.float64(-3.5383407969933938), pvalue=np.float64(0.007125697852423989), df=np.float64(8.353880491389793))


In [ ]:
t = -3.5383
df = 8.35
p_value = 2 * (1 - stats.t.cdf(abs(t), df))
print(p_value)

0.007131430640417991


Nivel de Significancia: 0.05

Decision: p<0.05: Hay Diferencia Significa entre las dos dietas.


# ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.



## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

Hipotesis Nula: Todos los fertilizantes producen el mismo rendimiento promedio.
Hipotesis Alternativa: Al menos uno de los fertilizantes produce un rendimiento diferente.

In [ ]:
import scipy.stats as stats
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import numpy as np

fertilizer_1 = [20, 21, 20, 19, 20]
fertilizer_2 = [22, 21, 23, 22, 21]
fertilizer_3 = [24, 23, 22, 23, 24]

# ANOVA test
f_value, p_value = stats.f_oneway(fertilizer_1, fertilizer_2, fertilizer_3)

print(f"f-value: {f_value}")
print(f"p-value: {p_value}")

f-value: 20.31578947368421
p-value: 0.000140478247931904


Nivel de Significancia: 0.05

Decision: p<0.05: Hay Diferencia Significa entre los Fertilizantes


In [ ]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import numpy as np

data = np.array(fertilizer_1 + fertilizer_2 + fertilizer_3)
labels = ['Fert1']*5 + ['Fert2']*5 + ['Fert3']*5

tukey = pairwise_tukeyhsd(data, labels, alpha=0.05)
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower  upper  reject
--------------------------------------------------
 Fert1  Fert2      1.8 0.0099 0.4572 3.1428   True
 Fert1  Fert3      3.2 0.0001 1.8572 4.5428   True
 Fert2  Fert3      1.4 0.0409 0.0572 2.7428   True
--------------------------------------------------


Media Rendimiento

In [ ]:
mean1 = np.mean(fertilizer_1)
mean2 = np.mean(fertilizer_2)
mean3 = np.mean(fertilizer_3)
print(mean1)
print(mean2)
print(mean3)

20.0
21.8
23.2


El Fertilizante 3 tiene la mayor media de rendimiento sobre el Fertilizante 1 y 2.  Y el Fertilizante 3 apoyado en la Media es 23.3 mayor que 20 y 21.8